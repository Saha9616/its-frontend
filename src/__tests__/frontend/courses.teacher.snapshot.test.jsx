import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import MyCourseView from "@/app/courses/page";

// mock useRouter
jest.mock("next/navigation", () => ({
  useRouter() {
    return {
      prefetch: () => null,
    };
  },
}));

// mock server action getUserProps
jest.mock("@/actions/getUserProps", () => ({
  getUserProps: jest.fn().mockResolvedValue({
    id: "mockId",
    email: "mockEmail",
    password: "mockPw",
    role: "TEACHER",
    school_id: "mockSchoolId",
    created_courses: [
      {
        id: "mockCourseId1",
        code: "mockCourseCode1",
        name: "mockCourseName1",
        creator_id: "mockId",
        school_id: "mockSchoolId",
      },
    ],
    joined_courses: [
      {
        id: "mockCourseId2",
        code: "mockCourseCode2",
        name: "mockCourseName2",
        creator_id: "mockCreatorId",
        school_id: "mockSchoolId",
      },
    ],
  }),
}));

it("renders teacher's courses page unchanged", async () => {
  const result = await MyCourseView();
  const { container } = render(result);
  expect(container).toMatchSnapshot();
});
