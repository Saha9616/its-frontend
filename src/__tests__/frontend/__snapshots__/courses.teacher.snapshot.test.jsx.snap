// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders teacher's courses page unchanged 1`] = `
<div>
  <div
    class="flex justify-between items-center"
  >
    <h1
      class="text-xl font-semibold"
    >
      My Courses
    </h1>
    <button
      aria-controls="radix-:r0:"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
      data-state="closed"
      type="button"
    >
      New Course
    </button>
  </div>
  <div
    class="mt-4 p-4 border rounded-md min-h-[calc(100vh-160px)]"
  >
    <div
      class="grid grid-cols-4 md:grid-cols-6 gap-4"
    >
      <a
        href="/courses/mockCourseCode1"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm h-40"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight flex items-center space-x-2"
            >
              <span>
                mockCourseCode1
              </span>
              <svg
                class="lucide lucide-clipboard-check "
                fill="none"
                height="17"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                style="margin-bottom: 2px;"
                viewBox="0 0 24 24"
                width="17"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="4"
                  rx="1"
                  ry="1"
                  width="8"
                  x="8"
                  y="2"
                />
                <path
                  d="M16 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H6a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2"
                />
                <path
                  d="m9 14 2 2 4-4"
                />
              </svg>
            </h3>
          </div>
          <div
            class="p-6 pt-0 space-y-2"
          >
            <h4
              class="font-medium"
            >
              mockCourseName1
            </h4>
          </div>
        </div>
      </a>
      <a
        href="/courses/mockCourseCode2"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm h-40"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight flex items-center space-x-2"
            >
              <span>
                mockCourseCode2
              </span>
            </h3>
          </div>
          <div
            class="p-6 pt-0 space-y-2"
          >
            <h4
              class="font-medium"
            >
              mockCourseName2
            </h4>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
